src/azul/service/elasticsearch_service.py

```diff
@@ -360,16 +358,10 @@ def _generate_paging_dict(self, es_response, pagination, sort_facet):
                 previous['search_before'], previous['search_before_uid'] = search_before
             else:
                 search_before = [None, None]
-            pagination.pop('search_after', None)
-            pagination.pop('search_after_uid', None)
         sort = pagination.pop('sort')
         self_url = pagination.pop('self_url')
-        next_ = self_url + '?' + urlencode(dict(sort=sort_facet,
-                                                **pagination,
-                                                **next_)) if next_ else None
-        previous = self_url + '?' + urlencode(dict(sort=sort_facet,
-                                                   **pagination,
-                                                   **previous)) if previous else None
+        next_ = self.page_link(next_, pagination, self_url, sort_facet)
+        previous = self.page_link(previous, pagination, self_url, sort_facet)
```

Comment:
This is definitely an improvement.

I think you could take it a bit further by changing `page_link()`'s signature to look like

```
def page_links(self, search_before, search_after, pagination, sort_facet) -> Tuple[str, str]:
```
and have it return `next_` and `prev` both. If you do this, then there's no need to assign any `next_` nor `prev` variables earlier, but you would need to check if `search_after` or `search_before` were none when building the link.

Just an idea to consider, but the code looks good overall.